Base/WebService.swift

```diff
@@ -12,38 +12,147 @@ public protocol UnauthorizedResponseHandler {
     func authorizedRequestDidFail(request: URLRequest, response: HTTPURLResponse, data: Data?, retry: @escaping () -> Void)
 }
 
+struct DataTaskCallback<A> {
+    let success: (URLResponse?, A) -> Void
+    let failure: (Error) -> Void
+    let retry: () -> Void
+    var unauthorizedResponseHandler: UnauthorizedResponseHandler?
+}
+
+extension URLSession {
+    
+    private func isCancelledError(_ error: NSError) -> Bool {
+        return error.domain == NSURLErrorDomain && error.code == NSURLErrorCancelled
+    }
+    
+    private func isNoConnectivityError(_ error: NSError) -> Bool {
+        return error.domain == NSURLErrorDomain && error.code == NSURLErrorNotConnectedToInternet
+    }
+    
+    fileprivate func handleErrorFor<A>(
```

Comment:
why file private ?

Reply:
This is the default for extract method on Xcode refactor, will update